### PR TITLE
[#86] Add support for providing `--data-dir` to CLI

### PR DESCRIPTION
**Pull request by AloeareV**
In browsing the various forks of `zecwallet-light-cli`, this looks like useful functionality to have!

### CODE REVIEW

File: lib/src/lightclient/lightclient_config.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ impl LightClientConfig {
                 monitor_mempool: true,
                 sapling_activation_height: info.sapling_activation_height,
                 anchor_offset: ANCHOR_OFFSET,
-                data_dir: None,
+                data_dir: data_dir,
```
Comment (zancas) [inline lib/src/lightclient/lightclient_config.rs:84]: Interesting, so there was a field in the struct, it just wasn't used.  Nice tweak.

**Review by zancas — APPROVED** [review]

File: lib/Cargo.toml [hunk 2]
```diff
@@ -48,9 +48,9 @@ group = "0.8"
 
 rust-embed = { version = "5.1.0", features = ["debug-embed"] }
 
-zcash_primitives = { git = "https://github.com/adityapk00/librustzcash", rev = "adeb3ec4ad15480482bc2962bc9fe453814db9ee", features = ["transparent-inputs"] }
-zcash_client_backend = { git = "https://github.com/adityapk00/librustzcash", rev = "adeb3ec4ad15480482bc2962bc9fe453814db9ee"}
-zcash_proofs = { git = "https://github.com/adityapk00/librustzcash", rev = "adeb3ec4ad15480482bc2962bc9fe453814db9ee", features = ["multicore"]}
+zcash_primitives = { git = "https://github.com/adityapk00/librustzcash", rev = "b70edf39e26e4c687b5087daf2ffd40f99fffd67", features = ["transparent-inputs"] }
```
Comment (zancas) [inline lib/Cargo.toml:51]: This looks like it needs to be rebased,   We shouldn't revert to pointing at legacy `librustzcash`.

**Review by zancas — CHANGES_REQUESTED** [review]
Points at legacy `librustzcash`.

**Comment by AloeareV** [comment]
Closing in favour of #64

**Pull request closed without merging**